### PR TITLE
feat: add configurable maximum search results setting for codebase indexing (#5149)

### LINKED ISSUE #5149: Add configurable maximum search results setting for codebase indexing
## What specific problem does this solve?

When using codebase indexing with the `codebase_search` function, users are currently limited to a maximum of 50 search results, which is hardcoded in the system. This limitation affects users who work with large codebases or need more comprehensive search results for their analysis.

**Who is affected:** All users who utilize codebase indexing functionality, particularly those working with large projects or requiring extensive search coverage.

**When this happens:** Every time users perform a codebase search operation through the indexing feature.

**Current behavior:** The `codebase_search` function returns a maximum of 50 results, regardless of user preference or project needs.

**Expected behavior:** Users should be able to configure the maximum number of search results returned by `codebase_search` through a slider control in the codebase indexing settings, allowing them to increase or decrease the limit based on their specific requirements.

**Impact:** Users with large codebases may miss relevant code sections due to the 50-result limit, while users with smaller projects might prefer fewer results for faster processing. The fixed limit reduces flexibility and may impact productivity for users who need more comprehensive search coverage.

## Additional context

This feature request is inspired by similar slider-based configuration functionality. Credit for this idea goes to @seedlord, who implemented something similar in https://github.com/seedlord/Roo-Code/tree/feat/codeindex-settings.

### PR DESCRIPTION
**Pull request by SannidhyaSah**
## Description

Fixes #5149

This PR implements a configurable maximum search results setting for codebase indexing, allowing users to control the number of search results returned through a slider in the settings UI.

## Changes Made

- **Core Configuration**: Added \maxSearchResults\ property to codebase index configuration interfaces with default value of 50
- **Settings UI**: Implemented a slider component in CodeIndexSettings.tsx with range 10-200 and step of 10
- **Search Integration**: Updated search service and Qdrant client to respect the configured limit
- **Type Safety**: Enhanced TypeScript interfaces across vector store and configuration modules
- **Comprehensive Testing**: Added unit tests for all new functionality including edge cases
- **Internationalization**: Added translations for the new setting in all 17 supported languages

Screenshot 2025-07-02 181139
 
## Testing

- [x] All existing tests pass
- [x] Added comprehensive unit tests for new functionality:
  - Configuration manager tests for maxSearchResults handling
  - Qdrant client tests for limit parameter usage
  - CodeIndexSettings component tests for slider behavior
- [x] Manual testing completed:
  - Verified slider appears in settings UI
  - Confirmed default value of 50 is applied
  - Tested range validation (10-200)
  - Verified search results respect the configured limit

## Verification of Acceptance Criteria

- [x] **Configurable limit**: Users can now set maximum search results via settings slider
- [x] **Default value**: Maintains backward compatibility with default of 50 results
- [x] **Range validation**: Enforces reasonable limits (10-200) to prevent performance issues
- [x] **UI Integration**: Seamlessly integrated into existing settings interface
- [x] **Type safety**: Full TypeScript support with proper interfaces

## Checklist

- [x] Code follows project style guidelines
- [x] Self-review completed
- [x] Comments added for complex logic
- [x] Documentation updated (if needed)
- [x] No breaking changes (or documented if any)
- [x] Accessibility checked (for UI changes)
- [x] All translations provided for 17 languages

----

> [!IMPORTANT]
> Adds configurable maximum search results setting for codebase indexing with UI slider, search integration, and comprehensive testing.
> 
>   - **Behavior**:
>     - Adds `codebaseIndexSearchMaxResults` to `codebase-index.ts` with a default of 50.
>     - Implements a slider in `CodeIndexSettings.tsx` for setting max results (range 10-200).
>     - Updates search service and Qdrant client to respect max results.
>   - **Testing**:
>     - Adds unit tests for `config-manager.spec.ts`, `qdrant-client.spec.ts`, and `CodeIndexSettings.spec.tsx`.
>     - Tests cover configuration handling, limit parameter usage, and slider behavior.
>   - **Internationalization**:
>     - Adds translations for new setting in 17 languages.
>   - **Misc**:
>     - Renames `MAX_SEARCH_RESULTS` to `DEFAULT_MAX_SEARCH_RESULTS` in `constants/index.ts`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=RooCodeInc%2FRoo-Code&utm_source=github&utm_medium=referral)<sup> for 90bd7855f394973f0582ed023442989a87065192. You can [customize](https://app.ellipsis.dev/RooCodeInc/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by seedlord** [comment]
https://github.com/RooCodeInc/Roo-Code/pull/5153 duplicate

**Comment by SannidhyaSah** [comment]
> https://github.com/RooCodeInc/Roo-Code/pull/5153 duplicate

Ignore that for now . The PR is incomplete.

File: src/services/code-index/vector-store/qdrant-client.ts [hunk 1]
```diff
@@ -297,7 +300,7 @@ export class QdrantVectorStore implements IVectorStore {
 				query: queryVector,
 				filter,
 				score_threshold: SEARCH_MIN_SCORE,
```
Comment (daniel-lxs) [inline src/services/code-index/vector-store/qdrant-client.ts]: I noticed that `score_threshold` is always set to `SEARCH_MIN_SCORE` here, but then conditionally overridden on line 314 if `minScore` is provided. This means the user-configured threshold from `configManager.currentSearchMinScore` (passed as the `minScore` parameter) might not be properly applied.

Could we simplify this by using:
```typescript
score_threshold: minScore ?? SEARCH_MIN_SCORE,
```

This would ensure the configured threshold is used when provided, falling back to the constant only when undefined.

File: src/services/code-index/constants/index.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ export const MAX_CHARS_TOLERANCE_FACTOR = 1.15 // 15% tolerance for max chars
 
 /**Search */
 export const SEARCH_MIN_SCORE = 0.4
-export const MAX_SEARCH_RESULTS = 50 // Maximum number of search results to return
+export const DEFAULT_MAX_SEARCH_RESULTS = 50 // Default maximum number of search results to return
```
Comment (daniel-lxs) [inline src/services/code-index/constants/index.ts:9]: I noticed the naming pattern inconsistency here. You've renamed `MAX_SEARCH_RESULTS` to `DEFAULT_MAX_SEARCH_RESULTS` (which is good!), but `SEARCH_MIN_SCORE` doesn't follow the same pattern.

For consistency, should we consider renaming `SEARCH_MIN_SCORE` to `DEFAULT_SEARCH_MIN_SCORE`? This would make it clearer that both are default values that can be overridden by user configuration.

**Review by daniel-lxs — COMMENTED** [review]
Hey @SannidhyaSah I left a couple of minor suggestions, I think this implementations is solid!

Let me know if you have any questions!

**Comment by MuriloFP** [comment]
Closed due to merging issues and moved to the new PR at #5402
